src/test/scala/chiselTests/ProbeSpec.scala

```diff
@@ -659,6 +659,8 @@ class ProbeSpec extends ChiselFlatSpec with MatchesAndOmits with Utils {
       define(b.refs.reg, RWProbeValue(r))
     }
     runTester(new BasicTester {
+      layer.enable(layers.Verification)
+      layer.enable(layers.Verification.Assert)
```

Comment:
Without these, the movement of the assertions into bound modules will cause this test to fail. This test is _extremely_ brittle because there are both forces and asserts and these need to happen in a very specific order for everything to work. As soon as the always blocks that this code was producing are fractured, then the order of things is no longer deterministic. This may speak to `force` being a footgun of a Chisel API or something that needs to be reconsidered as it is extremely procedural. `force_initial` is probably fine, though.

I did not try this test with `-extract-test-code`. I expect that would produce the same problems.

Reply:
Apologies for the brittle test, it does indeed excessively rely on statement ordering. 😞 .

Empirically, sinking the asserts into a child instance appears to resolve the dependence on its ordering.
I wonder why bound-in child instance behaves differently?  Hopefully the spec makes this clear...

All verif-y things (including printf, assert, so on) are entirely unordered re:Chisel and FIRRTL semantics and any reliance on their ordering is problematic and will cause problems in practice today, I'm not sure this is an issue with `force` specifically.  This has always been a problem but so far we haven't had to care.

`force` definitely is weird, though, and alternatives for use cases that reach for force seem like a great idea to iterate on :+1: for when that's not explicitly what's required.  This would help with flows not going to SV!

A simple test with prints and stops will be emitted out-of-order (printf/stop/printf), even without layers, for example.

Increasing usage of procedural statements in FIRRTL will continue to expose the unreliability of their ordering until we develop mechanisms to order them predictably / as needed or find means to not rely on their ordering.

Thanks for wrangling this test!